internal/provider/service_resource.go

```diff
@@ -381,6 +391,21 @@ func (r *ServiceResource) Create(ctx context.Context, req resource.CreateRequest
 	}
 }
 
+func (r *ServiceResource) attachNewServiceToVPC(ctx context.Context, plan serviceResourceModel, service *tsClient.Service) (*tsClient.Service, error) {
+	// Add service to vpc if vpc_id is provided
+	if plan.VpcID.IsNull() {
+		return service, nil
+	}
+
+	err := r.client.AttachServiceToVPC(ctx, service.ID, plan.VpcID.ValueInt64())
```

Comment:
In theory _service_ should not be nil here per line 359 (as service == nil -> error != nil -> we return on line 368) but every time I see a pointer dereferenced in a function without the nil check I get a bit scared. 🙂